src/SFA.DAS.EAS.Portal.UnitTests/Application/Commands/Cohort/CohortApprovalRequestedCommandHandlerTests.cs

```diff
@@ -0,0 +1,220 @@
+                a.Account.Organisations.First().Cohorts.Count.Equals(1)), It.IsAny<CancellationToken>()),
+                Times.Once);
+            }
+
+
+            [Test]
+            public async Task WhenANewCohortIsAdded_ThenApprentishipsInTheCohortAreStored()
+            {
+                // arrange
+                var testContext = new TestContext();
+                string cohortReference = Guid.NewGuid().ToString();
+                long apprenticeshipId = 123;
+
+                testContext.TestCommitment.Reference = cohortReference;
+                testContext.TestCommitment.Apprenticeships = new List<Commitments.Api.Types.Apprenticeship.Apprenticeship>() { new Commitments.Api.Types.Apprenticeship.Apprenticeship { Id = apprenticeshipId } };
+
+                CohortApprovalRequestedCommand command = new CohortApprovalRequestedCommandBuilder();
+
+                // act
+                await testContext.Sut.Handle(command);
+
+                //assert
+                testContext.MockAccountsService.Verify(m =>
+                m.Save(It.Is<AccountDocument>(a =>
+                a.Account.Organisations.First().Cohorts.Count.Equals(1) &&
+                a.Account.Organisations.First().Cohorts.First().Apprenticeships.Count.Equals(1) &&
+                a.Account.Organisations.First().Cohorts.First().Apprenticeships.First().Id.Equals(apprenticeshipId)), It.IsAny<CancellationToken>()),
+                Times.Once);
+            }
+
+            [Test]
+            public async Task WhenAnExistingCohortIsHandled_ThenAnyApprentishipsChangesAreAlsoStored()
+            {
+                // arrange
+                var testContext = new TestContext();
+                string cohortReference = Guid.NewGuid().ToString();
+                Apprenticeship apprenticeship = new ApprenticeshipBuilder();
+                Types.Cohort cohort = new CohortBuilder()
+                    .WithId(cohortReference)
+                    .WithApprenticeship(apprenticeship);
+                testContext.TestAccount.Organisations.First().Cohorts.Add(cohort);
+                testContext.TestAccount.Organisations.First().Cohorts.Count.Should().Be(1);
+
+                testContext.TestCommitment.Reference = cohortReference;
+                var testApprenticeship = new Commitments.Api.Types.Apprenticeship.Apprenticeship
+                {
+                    Id = apprenticeship.Id,
+                    FirstName = Guid.NewGuid().ToString(),
+                    LastName = Guid.NewGuid().ToString(),
+                    Cost = 123.45M,
+                    StartDate = DateTime.Now,
+                    EndDate = DateTime.Now.AddDays(10),
+                    TrainingName = Guid.NewGuid().ToString()
+                };
+                testContext.TestCommitment.Apprenticeships.Add(testApprenticeship);
+
+                testContext.TestAccount.Organisations.First().Cohorts.First().Apprenticeships.Count.Should().Be(1);
```

Comment:
We should assert the hierarchy, either using FluentAssertions collection asserts, or asserting each branch separately, so that if the graph doesn't match what is expected, an Assert is failed, rather than throwing e.g. a NullReferenceException.